<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SendRecvDialog</class>
 <widget class="QDialog" name="SendRecvDialog">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>817</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>Record Signal</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_4">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,10">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="9" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_sniff_Noise">
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="lineEditIP">
           <property name="text">
            <string>192.168.10.2</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QSpinBox" name="spinBoxGain">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="value">
            <number>40</number>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="labelGain">
           <property name="text">
            <string>Gain:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="labelBandWidth">
           <property name="text">
            <string>Bandwidth (Hz):</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="cbDevice">
           <item>
            <property name="text">
             <string>USRP</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>HackRF</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Device:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="labelSampleRate">
           <property name="text">
            <string>Sample rate (Sps):</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelIP">
           <property name="text">
            <string>IP address:</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QSpinBox" name="spinBoxNRepeat">
           <property name="specialValueText">
            <string>Infinite</string>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelFreq">
           <property name="text">
            <string>Frequency (Hz):</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="labelNRepeat">
           <property name="text">
            <string>Repeat:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="KillerDoubleSpinBox" name="spinBoxFreq">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="minimum">
            <double>0.001000000000000</double>
           </property>
           <property name="maximum">
            <double>1000000000000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="KillerDoubleSpinBox" name="spinBoxSampleRate">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="minimum">
            <double>0.001000000000000</double>
           </property>
           <property name="maximum">
            <double>1000000000000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="KillerDoubleSpinBox" name="spinBoxBandwidth">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="minimum">
            <double>0.001000000000000</double>
           </property>
           <property name="maximum">
            <double>1000000000000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="2" rowspan="2">
          <widget class="QToolButton" name="btnLockBWSR">
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../urh.qrc">
             <normaloff>:/icons/data/icons/lock.svg</normaloff>:/icons/data/icons/lock.svg</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_sniff_Center">
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>-3.140000000000000</double>
           </property>
           <property name="maximum">
            <double>3.140000000000000</double>
           </property>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="label_sniff_Noise">
           <property name="text">
            <string>Noise:</string>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="label_sniff_Center">
           <property name="text">
            <string>Center:</string>
           </property>
          </widget>
         </item>
         <item row="11" column="0">
          <widget class="QLabel" name="label_sniff_Modulation">
           <property name="text">
            <string>Modulation:</string>
           </property>
          </widget>
         </item>
         <item row="11" column="1">
          <widget class="QComboBox" name="combox_sniff_Modulation">
           <item>
            <property name="text">
             <string>ASK</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>FSK</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>PSK</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="13" column="0">
          <widget class="QLabel" name="label_sniff_OutputFile">
           <property name="text">
            <string>Output file:</string>
           </property>
          </widget>
         </item>
         <item row="13" column="1">
          <widget class="QLineEdit" name="lineEdit_sniff_OutputFile">
           <property name="readOnly">
            <bool>false</bool>
           </property>
           <property name="placeholderText">
            <string>None</string>
           </property>
           <property name="clearButtonEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_sniff_BitLength">
           <property name="text">
            <string>Bit Length:</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QSpinBox" name="spinbox_sniff_BitLen">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
         <item row="10" column="0">
          <widget class="QLabel" name="label_sniff_Tolerance">
           <property name="text">
            <string>Error Tolerance:</string>
           </property>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QSpinBox" name="spinbox_sniff_ErrorTolerance">
           <property name="maximum">
            <number>999999</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="12" column="1">
          <widget class="QComboBox" name="comboBox_sniff_viewtype">
           <item>
            <property name="text">
             <string>Bit</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Hex</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>ASCII</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="12" column="0">
          <widget class="QLabel" name="label_sniff_viewtype">
           <property name="text">
            <string>View:</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="topMargin">
           <number>0</number>
          </property>
          <item row="14" column="0">
           <widget class="QProgressBar" name="progressBar">
            <property name="value">
             <number>0</number>
            </property>
            <property name="format">
             <string>%v/%m</string>
            </property>
           </widget>
          </item>
          <item row="13" column="0">
           <widget class="QLabel" name="lSamplesSentText">
            <property name="text">
             <string>Samples sent:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>10</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="lTimeText">
            <property name="font">
             <font>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="text">
             <string>Time (in seconds):</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lSamplesCapturedText">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="text">
             <string>Samples captured:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QToolButton" name="btnStart">
              <property name="minimumSize">
               <size>
                <width>42</width>
                <height>42</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>42</width>
                <height>42</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Record signal</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset theme="media-record">
                <normaloff>.</normaloff>.</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>32</width>
                <height>32</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btnStop">
              <property name="minimumSize">
               <size>
                <width>42</width>
                <height>42</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>42</width>
                <height>42</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Stop recording</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset theme="media-playback-stop">
                <normaloff>.</normaloff>.</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>32</width>
                <height>32</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btnSave">
              <property name="minimumSize">
               <size>
                <width>42</width>
                <height>42</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>42</width>
                <height>42</height>
               </size>
              </property>
              <property name="text">
               <string>Save...</string>
              </property>
              <property name="icon">
               <iconset theme="document-save">
                <normaloff>.</normaloff>.</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>32</width>
                <height>32</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btnClear">
              <property name="minimumSize">
               <size>
                <width>42</width>
                <height>42</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>42</width>
                <height>42</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Clear</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset theme="view-refresh">
                <normaloff>.</normaloff>.</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>32</width>
                <height>32</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="lSignalSizeText">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="text">
             <string>Signal size (in MiB):</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QLabel" name="lSamplesCaptured">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="10" column="0" colspan="2">
           <widget class="QLabel" name="lTime">
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="6" column="0" colspan="2">
           <widget class="QLabel" name="lSignalSize">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="lblRepeatText">
            <property name="text">
             <string>Current iteration:</string>
            </property>
           </widget>
          </item>
          <item row="12" column="0">
           <widget class="QLabel" name="lblCurrentRepeatValue">
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="txtEditErrors">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QStackedWidget" name="stackedWidget">
         <property name="currentIndex">
          <number>2</number>
         </property>
         <widget class="QWidget" name="page_receive">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="LiveGraphicView" name="graphicsViewReceive">
             <property name="verticalScrollBarPolicy">
              <enum>Qt::ScrollBarAlwaysOff</enum>
             </property>
             <property name="horizontalScrollBarPolicy">
              <enum>Qt::ScrollBarAsNeeded</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_send">
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="EditableGraphicView" name="graphicsViewSend">
             <property name="mouseTracking">
              <bool>true</bool>
             </property>
             <property name="renderHints">
              <set>QPainter::Antialiasing|QPainter::TextAntialiasing</set>
             </property>
             <property name="transformationAnchor">
              <enum>QGraphicsView::NoAnchor</enum>
             </property>
             <property name="resizeAnchor">
              <enum>QGraphicsView::NoAnchor</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_7">
             <property name="font">
              <font>
               <italic>true</italic>
              </font>
             </property>
             <property name="text">
              <string>Hint: You can edit the raw signal before sending.</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page">
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QPlainTextEdit" name="txtEd_sniff_Preview">
             <property name="lineWrapMode">
              <enum>QPlainTextEdit::NoWrap</enum>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="maximumBlockCount">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnAccept">
             <property name="text">
              <string>Accept data</string>
             </property>
             <property name="autoDefault">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QLabel" name="label_y_scale">
           <property name="text">
            <string>Y-Scale</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="sliderYscale">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="tickInterval">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KillerDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>urh.ui.KillerDoubleSpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>LiveGraphicView</class>
   <extends>QGraphicsView</extends>
   <header>urh.ui.views.LiveGraphicView.h</header>
  </customwidget>
  <customwidget>
   <class>EditableGraphicView</class>
   <extends>QGraphicsView</extends>
   <header>urh.ui.views.EditableGraphicView.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>cbDevice</tabstop>
  <tabstop>lineEditIP</tabstop>
  <tabstop>spinBoxFreq</tabstop>
  <tabstop>spinBoxSampleRate</tabstop>
  <tabstop>spinBoxBandwidth</tabstop>
  <tabstop>spinBoxGain</tabstop>
  <tabstop>spinBoxNRepeat</tabstop>
  <tabstop>btnStart</tabstop>
  <tabstop>btnStop</tabstop>
  <tabstop>btnSave</tabstop>
  <tabstop>btnClear</tabstop>
  <tabstop>sliderYscale</tabstop>
  <tabstop>txtEditErrors</tabstop>
 </tabstops>
 <resources>
  <include location="../urh.qrc"/>
 </resources>
 <connections/>
</ui>
